#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.11092445161906588 -0.9577176901047024 -0.26546674385602903 2.3802706905999953
  position -0.48563020420556685 0.17575311838479246 0.5311346064256073
  follow "robot"
}
DirectionalLight {
  ambientIntensity 1
  direction 0 -1 0
}
DirectionalLight {
  direction 1 -1 0
}
DirectionalLight {
  direction -1 -1 0
}
DirectionalLight {
  direction 0 -1 -1
}
DirectionalLight {
  direction 0 -1 1
}
RectangleArena {
  translation 0.12 0 -0.28
  floorSize 3 3
  floorTileSize 0.25 0.25
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
}
Solid {
  translation -0.4 0 -1.63
  scale 0.001 0.001 0.001
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "first_part.STL"
        ]
      }
    }
  ]
  locked TRUE
}
DEF Team Robot {
  translation -0.230002 0 0.67
  rotation 0 1 0 -1.5707996938995747
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.411765 0 0.411765
        roughness 1
        metalness 0
      }
      geometry DEF Body IndexedFaceSet {
        coord Coordinate {
          point [
            0 0.005 0
            0.15 0.005 0
            0 0.005 0.1
            0.15 0.005 0.1
            0 0.008 0.003
            0 0.008 0.097
            0.147 0.008 0.003
            0.147 0.008 0.097
            0 0.025 0
            0.15 0.025 0
            0 0.025 0.1
            0.15 0.025 0.1
            0 0.025 0.003
            0 0.025 0.097
            0.147 0.025 0.003
            0.147 0.025 0.097
          ]
        }
        ccw FALSE
        convex FALSE
        coordIndex [
          0, 2, 3, 1, -1, 0, 1, 9, 8, -1, 2, 10, 11, 3, -1, 3, 11, 9, 1, -1, 5, 4, 6, 7, -1, 4, 12, 14, 6, -1, 5, 7, 15, 13, -1, 7, 6, 14, 15, -1, 0, 8, 12, 4, 5, 13, 10, 2, -1, 8, 9, 11, 10, 13, 15, 14, 12, -1
        ]
      }
    }
    DEF RB HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.14 0.015 -0.0075
      }
      device [
        PositionSensor {
          name "right_back_encoder"
          resolution 0.0028
        }
        RotationalMotor {
          name "right_back_motor"
          maxVelocity 22
          maxTorque 0.848
        }
      ]
      endPoint Solid {
        translation 0.14 0.0150004 -0.0075
        rotation -1 -1.154909002864889e-12 1.1541828363252895e-12 1.5707996939090054
        children [
          Shape {
            appearance DEF Track_wheel_black PBRAppearance {
              baseColor 0.15 0.15 0.15
              roughness 1
              metalness 0
            }
            geometry DEF Track_wheel_geo Cylinder {
              height 0.01
              radius 0.015
            }
          }
        ]
        name "right_back_wheel"
        boundingObject USE Track_wheel_geo
        physics Physics {
          density -1
          mass 0.02
        }
      }
    }
    DEF RF HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.01 0.015 -0.0075
      }
      device [
        RotationalMotor {
          name "right_front_motor"
          maxVelocity 22
          maxTorque 0.848
        }
      ]
      endPoint Solid {
        translation 0.010000084947586363 0.015000331480982648 -0.007500000185799181
        rotation -1 -6.763063749008397e-13 6.756194606066238e-13 1.570799693908768
        children [
          Shape {
            appearance USE Track_wheel_black
            geometry USE Track_wheel_geo
          }
        ]
        name "right_front_wheel"
        boundingObject USE Track_wheel_geo
        physics Physics {
          density -1
          mass 0.02
        }
      }
    }
    DEF LB HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.14 0.015 0.1075
      }
      device [
        PositionSensor {
          name "left_back_encoder"
          resolution 0.0028
        }
        RotationalMotor {
          name "left_back_motor"
          maxVelocity 22
          maxTorque 0.848
        }
      ]
      endPoint Solid {
        translation 0.14 0.0149999 0.1075
        rotation -1 1.5828153929799536e-11 1.5587380656664818e-10 1.5707996943179932
        children [
          Shape {
            appearance USE Track_wheel_black
            geometry USE Track_wheel_geo
          }
        ]
        name "left_back_wheel"
        boundingObject USE Track_wheel_geo
        physics Physics {
          density -1
          mass 0.02
        }
      }
    }
    DEF LF HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.01 0.015 0.1075
      }
      device [
        RotationalMotor {
          name "left_front_motor"
          maxVelocity 22
          maxTorque 0.848
        }
      ]
      endPoint Solid {
        translation 0.010000084944961964 0.01500038574195705 0.10749999981692183
        rotation -1 -5.207987917517216e-13 5.20092251334814e-13 1.5707996939087936
        children [
          Shape {
            appearance USE Track_wheel_black
            geometry USE Track_wheel_geo
          }
        ]
        name "left_front_wheel"
        boundingObject USE Track_wheel_geo
        physics Physics {
          density -1
          mass 0.02
        }
      }
    }
    DistanceSensor {
      translation 0 0.005 0.07
      rotation 0 0 1 -1.5707996938995747
      name "l2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.005 0.06
      rotation 0 0 1 -1.5707996938995747
      name "l1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.005 0.04
      rotation 0 0 1 -1.5707996938995747
      name "r1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.005 0.03
      rotation 0 0 1 -1.5707996938995747
      name "r2"
      type "infra-red"
    }
  ]
  boundingObject USE Body
  physics Physics {
    density -1
    mass 0.1
  }
  controller "Shortie_line_follow"
}
